src/utils/filesystem.posix.cc

```diff
@@ -84,4 +84,43 @@ time_t GetFileTime(const std::string& filename) {
 #endif
 }
 
+std::string GetUserCacheDirectory() {
+#ifdef __APPLE__
+  constexpr auto kLocalDir = "/Library/Caches";
```

Comment:
Nitpicking:
here it looks like those would be absolute directories from root, maybe write it like `"Library/Caches"` here and `std::string(home) + "/" + kLocalDir` below?

Also, maybe it's time to switch to C++17 and use https://en.cppreference.com/w/cpp/filesystem/path/append ?

And also, I think it's better if it contains trailing slash. Then client code can do `GetUserCacheDirectory() + "lc0.config"` and then it will work "fine" if there's no `$HOME` defined (otherwise it will create config at `/`).

Reply:
Just to clarify, missing trailing slash seems to be a real problem (if calling code will do `GetDir() +"/" + "filename.txt"`, e.g. on windows, it will get resolved into `/filename.txt`, and not `./filename.txt`).

Other comments are totally optional.